modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/result/AbstractReturnDelegate.java

```diff
@@ -322,6 +356,9 @@ public void close() throws IOException {
         in = new ClosedInputStream();
       } finally {
         connectionHandler.release();
+        if (onClose != null && alreadyClosed.compareAndSet(false, true)) {
```

Comment:
this should be a try.finally of it's own.. .this needs to happen even if `connectionHandler.release()` fails